File: functions/src/models/game.ts
class Game {
  id: number;
  name: string;
  date: string;
  cover: string;
  platforms: string;
  description: string;
  website: string;
  screenshots: string[];

  constructor(
    id = 0,
    name = "",
    date = "",
    cover = "",
    platforms = "",
    description = "",
    website = "",
    screenshots: string[] = []
  ) {
    this.id = id;
    this.name = name;
    this.date = date;
    this.cover = cover;
    this.platforms = platforms;
    this.description = description;
    this.website = website;
    this.screenshots = screenshots;
  }
}

export = Game;

File: lib/models/game.dart
class Game {
  int id;
  String name;
  String date;
  String cover;
  String platforms;
  String description;
  String website;
  List<String> screenshots;

  Game({
    this.id = 0,
    this.name = "",
    this.date = "",
    this.cover = "",
    this.platforms = "",
    this.description = "",
    this.website = "",
    List<String>? screenshots,
  }) : screenshots = screenshots ?? <String>[];

  Game.fromJson(Map<dynamic, dynamic> json)
      : id = json['id'],
        name = json['name'],
        date = json['date'],
        cover = json['cover'],
        platforms = json['platforms'],
        description = json['description'],
        website = json['website'],
        screenshots = List<String>.from(json['screenshots'] ?? []);
}
